src/otx/algo/visual_prompting/openvino_models.py

```diff
@@ -0,0 +1,151 @@
+
+from copy import deepcopy
+from typing import TYPE_CHECKING, Any
+
+import numpy as np
+from openvino.model_api.models import ImageModel, SegmentationModel
+from openvino.model_api.models.types import BooleanValue, NumericalValue, StringValue
+
+if TYPE_CHECKING:
+    from openvino.model_api.adapters.inference_adapter import InferenceAdapter
+
+
+class ImageEncoder(ImageModel):
+    """Image Encoder class of OTX Visual Prompting model for openvino task."""
+
+    __model__ = "image_encoder"
+
+    def __init__(
+        self,
+        inference_adapter: InferenceAdapter,
+        configuration: dict[str, Any] | None = None,
+        preload: bool = False,
+    ):
+        super().__init__(inference_adapter, configuration, preload)
+
+    @classmethod
+    def parameters(cls) -> dict[str, Any]:  # noqa: D102
+        parameters = super().parameters()
+        parameters.update(
+            {
+                "resize_type": StringValue(default_value="fit_to_window"),
+                "image_size": NumericalValue(value_type=int, default_value=1024, min=0, max=2048),
+            },
+        )
+        return parameters
+
+    def preprocess(self, inputs: np.ndarray) -> tuple[dict[str, np.ndarray], dict[str, Any]]:
+        """Update meta for image encoder."""
+        dict_inputs, meta = super().preprocess(inputs)
+        meta["resize_type"] = self.resize_type
+        return dict_inputs, meta
+
+
+class Decoder(SegmentationModel):
+    """Decoder class of OTX Visual Prompting model for openvino task."""
+
+    __model__ = "decoder"
+
+    def __init__(
+        self,
+        model_adapter: InferenceAdapter,
+        configuration: dict | None = None,
+        preload: bool = False,
+    ):
+        super().__init__(model_adapter, configuration, preload)
+
+        self.mask_input = np.zeros((1, 1, 256, 256), dtype=np.float32)
+        self.has_mask_input = np.zeros((1, 1), dtype=np.float32)
+
+    @classmethod
+    def parameters(cls) -> dict[str, Any]:  # noqa: D102
+        parameters = super().parameters()
+        parameters.update({"image_size": NumericalValue(value_type=int, default_value=1024, min=0, max=2048)})
+        parameters.update({"mask_threshold": NumericalValue(value_type=float, default_value=0.0, min=0, max=1)})
+        parameters.update({"embed_dim": NumericalValue(value_type=int, default_value=256, min=0, max=512)})
+        parameters.update({"embedded_processing": BooleanValue(default_value=True)})
+        return parameters
+
+    def _get_outputs(self) -> str:
+        return "upscaled_masks"
+
+    def preprocess(self, inputs: dict[str, Any]) -> list[dict[str, Any]]:
+        """Preprocess prompts."""
+        processed_prompts: list[dict[str, Any]] = []
+        idx: int = 0
+        for prompt_name in ["bboxes", "points"]:
+            prompts = inputs.get(prompt_name, None)
+            if prompts is None:
+                continue
+            for prompt in prompts:
+                label = inputs["labels"][idx]
```

Comment:
does this mean all prompts have different label?

Reply:
They can be.

Reply:
Sorry, I am asking that this is intended or not. In the code, it looks like to enforce all should be different label.

Reply:
I intended. Each prompt has their own label and it can be same or different with each other.
https://github.com/openvinotoolkit/training_extensions/blob/2708d7ba16465e35cd7665da3043223ecc2dde3f/src/otx/core/data/dataset/visual_prompting.py#L121